src/system/game-stats.ts

```diff
@@ -58,9 +58,9 @@ export class GameStats {
     this.pokemonCaught = source?.pokemonCaught || 0;
     this.pokemonHatched = source?.pokemonHatched || 0;
     // Currently handled by migration
-    this.subLegendaryPokemonSeen = source?.subLegendaryPokemonSeen;
-    this.subLegendaryPokemonCaught = source?.subLegendaryPokemonCaught;
-    this.subLegendaryPokemonHatched = source?.subLegendaryPokemonHatched;
+    this.subLegendaryPokemonSeen = source?.subLegendaryPokemonSeen || 0;
+    this.subLegendaryPokemonCaught = source?.subLegendaryPokemonCaught || 0;
+    this.subLegendaryPokemonHatched = source?.subLegendaryPokemonHatched || 0;
```

Comment:
don't use `||`. If possible use `??`.
Some for the other ones.
cause with `||` the `0` is also considered `false`. But `??` only considers `null` or `undefined` as false

```suggestion
    this.subLegendaryPokemonSeen = source?.subLegendaryPokemonSeen ?? 0;
    this.subLegendaryPokemonCaught = source?.subLegendaryPokemonCaught ?? 0;
    this.subLegendaryPokemonHatched = source?.subLegendaryPokemonHatched ?? 0;
```